### PR TITLE
f-enhancement/gh-1021-group-audit

### PR DESCRIPTION
**Pull request by jprotic**
Resolves: #1021

### CODE REVIEW

File: src/main/frontend/src/components/audit/allowance/AllowanceAudits.js [hunk 1]
```diff
@@ -5,28 +5,59 @@ import AllowanceAudit from './AllowanceAudit';
...
+            size="15px"
+            src={ArrowIcon}
+            alt="Arrow for toggling accordion item"
           />
-        ))}
-        {audit.allowance.length === 0 && (
-          <div>This user has no allowance manual adjust edit history.</div>
+        </ExpandableAudit>
+        {isOpen && (
+          <div>
```
Comment (iscekic) [inline src/main/frontend/src/components/audit/allowance/AllowanceAudits.js]: Do we need a `div` or would a `Fragment` be enough?

File: src/main/frontend/src/components/audit/employee/EmployeeAudits.js [hunk 2]
```diff
@@ -9,32 +9,61 @@ import {
 } from '../../../store/selectors';
 import EmployeeAudit from './EmployeeAudit';
 import { DrawerContent } from '../../../pages/administration/components/common/DrawerElements';
+import { Icon } from '@/pages/requests/components/styled';
+import ArrowIcon from '../../../../src/img/icons/arrow-down.svg';
+import { ExpandableAudit } from '../ExpandableAudit';
 
 class EmployeeAudits extends Component {
```
Comment (iscekic) [inline src/main/frontend/src/components/audit/employee/EmployeeAudits.js]: `EmployeeAudits` and `AllowanceAudits` look very similar, would it make sense to extract some common functionality into a separate component?

File: src/main/frontend/src/components/audit/allowance/AllowanceAudits.js [hunk 3]
```diff
@@ -1,32 +1,58 @@
...
+      isOpen: !prevState.isOpen
+    }));
+  };
+
   render() {
     const { audit, leaveProfiles } = this.props;
 
+    const { isOpen } = this.state;
+
+    const type = 'Allowance';
```
Comment (iscekic) [inline src/main/frontend/src/components/audit/allowance/AllowanceAudits.js]: Redundant variable.

File: src/main/frontend/src/components/audit/employee/EmployeeAudits.js [hunk 4]
```diff
@@ -9,32 +9,56 @@ import {
...
   render() {
     const {
       audit,
       teams: { results: teams },
       leaveProfiles
     } = this.props;
 
+    const { isOpen } = this.state;
+
+    const type = 'Employee';
```
Comment (iscekic) [inline src/main/frontend/src/components/audit/employee/EmployeeAudits.js]: Redundant variable.

File: src/main/frontend/src/components/audit/ExpandableAudits.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  border: 1px solid lightgray;
+  border-radius: 5px;
+  width: 100%;
+  cursor: pointer;
+  text-align: center;
+  margin-bottom: 10px;
+  position: relative;
+`;
+
+class ExpandableAudits extends Component {
```
Comment (jolpazy) [inline src/main/frontend/src/components/audit/ExpandableAudits.js]: Can this be a functional component with props destructured in parameter? 
eg: 
const ExpandableAudits = ({ type, open }) => ...

File: src/main/frontend/src/components/audit/ExpandableAudits.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Icon } from '@/pages/requests/components/styled';
+import ArrowIcon from '../../../src/img/icons/arrow-down.svg';
```
Comment (jolpazy) [inline src/main/frontend/src/components/audit/ExpandableAudits.js]: lets try and use absolute imports, at least in new files (:

File: src/main/frontend/src/components/audit/employee/EmployeeAudits.js [hunk 7]
```diff
@@ -9,33 +9,54 @@ import {
 } from '../../../store/selectors';
 import EmployeeAudit from './EmployeeAudit';
 import { DrawerContent } from '../../../pages/administration/components/common/DrawerElements';
+import ExpandableAudits from '../ExpandableAudits';
+import { Collapse } from 'reactstrap';
```
Comment (jolpazy) [inline src/main/frontend/src/components/audit/employee/EmployeeAudits.js]: also, import React and libraries at top, then components at the bottom (just a nitpick)

**Pull request merged**